app/uk/gov/hmrc/securemessage/controllers/models/generic/ApiConversation.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.controllers.models.generic
+
+import cats.implicits._
+import cats.kernel.Eq
+import org.joda.time.DateTime
+import play.api.libs.json.{ Json, Writes }
+import uk.gov.hmrc.securemessage.models.core.{ Identifier, _ }
+
+final case class ApiConversation(
+  client: String,
+  conversationId: String,
+  status: ConversationStatus,
+  tags: Option[Map[String, String]],
+  subject: String,
+  language: Language,
+  messages: List[ApiMessage]
+)
+
+object ApiConversation {
+
+  def coreConversationToApiConversation(conversation: Conversation, identifier: Identifier): ApiConversation =
+    ApiConversation(
+      client = conversation.client,
+      conversationId = conversation.conversationId,
+      status = conversation.status,
+      tags = conversation.tags,
+      subject = conversation.subject,
+      language = conversation.language,
+      messages = conversation.messages.map(message => apiMessageDetails(conversation, message, identifier))
+    )
+
+  private def apiMessageDetails(
```

Comment:
could we maybe rename this? something like convertToApiMessage(...) or build... or constructApiMessage(...)?

Reply:
I'll change it now